prow/plank/controller.go

```diff
@@ -112,24 +158,31 @@ func (c *Controller) Sync() error {
 	for _, pod := range pods {
 		pm[pod.Metadata.Name] = pod
 	}
-	pending := pendingJobs(pjs)
+	c.updatePendingJobs(pjs)
 	var syncErrs []error
 	if err := c.terminateDupes(pjs); err != nil {
 		syncErrs = append(syncErrs, err)
 	}
+
+	pjCh := make(chan kube.ProwJob, len(pjs))
 	for _, pj := range pjs {
-		if pj.Spec.Agent == kube.KubernetesAgent {
-			if err := c.syncKubernetesJob(pj, pm, pending); err != nil {
-				syncErrs = append(syncErrs, err)
-			}
-		} else if pj.Spec.Agent == kube.JenkinsAgent {
-			if err := c.syncJenkinsJob(pj, pending); err != nil {
-				syncErrs = append(syncErrs, err)
-			}
-		} else {
-			syncErrs = append(syncErrs, fmt.Errorf("job %s has unsupported agent %s", pj.Metadata.Name, pj.Spec.Agent))
-		}
+		pjCh <- pj
 	}
+	close(pjCh)
+
+	errCh := make(chan error, len(pjs))
```

Comment:
When someone is refactoring, if they push more than one `error` per `ProwJob` into this they can deadlock the controller. Do we want a stronger guarantee that this won't happen?